concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/AccountReleaseSchedule.hs

```diff
@@ -34,113 +35,130 @@ module Concordium.GlobalState.Basic.BlockState.AccountReleaseSchedule (
-
-
-    where toObject :: (Timestamp, (Amount, [TransactionHash])) -> AE.Value
-          toObject (timestamp, (amount, hashes)) = AE.object [
-            "timestamp" AE..= timestamp,
-            "amount" AE..= amount,
-            "transactions" AE..= hashes
-            ]
+-- |Produce an 'AccountReleaseSummary' from an 'AccountReleaseSchedule'.
+toAccountReleaseSummary :: AccountReleaseSchedule -> AccountReleaseSummary
+toAccountReleaseSummary AccountReleaseSchedule{..} = AccountReleaseSummary{..}
+  where
+    releaseTotal = _totalLockedUpBalance
+    listOfReleasesByTimestamp = [(tm, (a, t)) | Just (r, t) <- Vector.toList _values, Release tm a <- r]
+    sortedAndGroupedByTimestamp = groupBy ((==) `on` fst) $ sortOn fst listOfReleasesByTimestamp
+    accumReleases (!accB, !accT) (_, (b, t)) = (accB + b, t : accT)
+    makeScheduledRelease rels =
+        let (releaseAmount, releaseTransactions) = foldl' accumReleases (0, []) rels
+         in ScheduledRelease
+                { -- @head@ is safe since group won't create empty lists
+                  releaseTimestamp = fst (head rels),
+                  ..
+                }
+    releaseSchedule = map makeScheduledRelease sortedAndGroupedByTimestamp
 
 instance Serialize AccountReleaseSchedule where
-  get = do
-    vecLength <- getLength
-    _values <- Vector.replicateM vecLength (getMaybe (do
-                                                         l <- getLength
-                                                         item <- sequence $ replicate l get
-                                                         txh <- get
-                                                         return (item, txh)))
-    let (_pendingReleases, _totalLockedUpBalance) =
-          Vector.ifoldl' (\acc idx -> \case
-                             Nothing -> acc
-                             Just (rel, _) ->
-                               let f (pending, am) Release{..} =
-                                     let pending' = Map.alter (maybe (Just [idx]) (Just . (idx :))) timestamp pending
-                                         am' = am + amount
+    get = do
+        vecLength <- getLength
+        _values <-
+            Vector.replicateM
+                vecLength
+                ( getMaybe
+                    ( do
+                        l <- getLength
+                        item <- replicateM l get
+                        txh <- get
+                        return (item, txh)
+                    )
+                )
+        let (_pendingReleases, _totalLockedUpBalance) =
+                Vector.ifoldl'
+                    ( \acc idx -> \case
+                        Nothing -> acc
+                        Just (rel, _) ->
+                            let f (pending, am) Release{..} =
+                                    let pending' = Map.alter (maybe (Just [idx]) (Just . (idx :))) timestamp pending
+                                        am' = am + amount
```

Comment:
I'm guessing only indendation/line breaks changed, but this format style seems substantially worse than the previous one.

Reply:
I don't love everything about the fourmolu formatting style, but I don't agree that it's substantially worse. I think it actually more clearly delineates the arguments of multi-line function calls such as `replicateM` and `ifoldl'` here.